src/Microsoft.ML.FastTree/FastTree.cs

```diff
@@ -151,7 +151,7 @@ public abstract class FastTreeTrainerBase<TArgs, TTransformer, TModel> :
         /// Constructor that is used when invoking the classes deriving from this, through maml.
         /// </summary>
         private protected FastTreeTrainerBase(IHostEnvironment env, TArgs args, SchemaShape.Column label)
-            : base(Contracts.CheckRef(env, nameof(env)).Register(RegisterName), TrainerUtils.MakeR4VecFeature(args.FeatureColumn), label, TrainerUtils.MakeR4ScalarWeightColumn(args.WeightColumn, args.WeightColumn.IsExplicit))
```

Comment:
What is/was `args.WeightColumn.IsExplicit` used for?

Reply:
it is something that got introduced with EntryPoints. If the user doesn't specify  a name for the optional columns, an Implict argument get created for them... not sure of the original thoughts around why not just leave it null. Maybe to avoid dealing with its serialization across languages.. idk. 

---
In reply to: [248127604](https://github.com/dotnet/machinelearning/pull/2158#discussion_r248127604) [](ancestors = 248127604)